src/components/navbar/navLinkSection/NavLinkSection.jsx

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+import classes from './NavLinkSection.module.scss';
+import PropTypes from 'prop-types';
+
+import { openNewsletterSignupModal } from '../.././../utils/functions';
+
+const NavLinkSection = ({ navLinks, toggleDisplayNav }) => {
+    const handleNewsletterClick = () => {
+        if (toggleDisplayNav) {
+            toggleDisplayNav(false);
+        }
+        openNewsletterSignupModal(true);
+    };
+
```

Comment:
maybe this?
```suggestion
closeDisplayNavOnMobile() {
  if(toggleDisplayNav) {
     toggleDisplayNav()
  }
}
```

Reply:
If the function is an optional argument, it has to be able to handle the function being null - you cant call null (hence the issue @stellenberger was having).